dbuturu/Store-Manager
# #161399129 user authentication

#### What does this PR do?
Gives an Admin or the creator (store attendant) the ability to get specific sale order records

#### Description of Task to be completed?
The following  endpoints were added:
/users/
* get
  * read all users
* post
  * create a user

 /users/login/
* post
  * login a user

 /users/logout/
* post
  * logout a user

 /users/{username}
* delete
  * delete specific user
* put
  * update specific user
* get
  * read all users

#### How should this be manually tested?
Clone this repository
git clone https://github.com/dbuturu/Store-Manager.git
Check out this branch.
git checkout ch-User_authentication-API-#161399129
Create a virtual environment
virtualenv venv
Active it by a run active command
source venv/bin/activate 
source venv/Scripts/activate 
venv\activate.bat 
Install dependencies
pip install -r requirements.txt
Start the application.
python run.py
Run Tests
pytest -v

#### Any background context you want to provide?
This is an API chore

#### What are the relevant pivotal tracker stories?
[#161399129](https://www.pivotaltracker.com/story/show/161399129)
